homework-submission/week5-homework/src/stores/TodoStore.js

```diff
@@ -0,0 +1,148 @@
+import { configure, observable, action, computed, runInAction } from "mobx";
+import uuidv4 from "uuid/v4";
+const url = "https://hyf-react-api.herokuapp.com";
+
+configure({ enforceActions: "observed" });
+class TodoClass {
+  @observable
+  todos = [];
+  @observable
+  state = "Loading";
+
+  @action
+  async getTodos() {
+    this.state = "Loading";
+    this.todos = [];
+    try {
+      let todos = await this.getData();
+      runInAction(() => {
+        this.state = "Done";
+        this.todos = todos;
+      });
+    } catch (error) {
+      runInAction(() => {
+        this.state = "error";
+      });
+    }
+  }
+
+  @action
+  toggleDone = async id => {
+    let todoItem = this.todos.find(todo => todo._id === id);
+    try {
+      await this.handleFetch(`${url}/todos/${id}`, "PATCH", {
+        done: !todoItem.done
+      });
+      runInAction(() => {
+        let todoItems = this.todos.map(todo => {
+          if (id === todo._id) {
+            return {
+              ...todo,
+              done: !todo.done
+            };
+          }
+          return todo;
+        });
+        this.todos = todoItems;
+      });
+    } catch (error) {
+      runInAction(() => {
+        this.state = "Error";
+      });
+    }
+  };
+
+  @action
+  addTodo = async newTodo => {
+    let todoItem = {
+      id: uuidv4(),
```

Comment:
no need for the `uuid` use here.
while you are dealing with a server this is just like an extra information you are sending to the server, the server is just handling this for us.